### PR TITLE
Mac OS X build

### PR DESCRIPTION
**Pull request by ea2973929**

### CODE REVIEW

**Comment by dvc94ch** [comment]
If you have time and interest in getting the ios build working, that would be awesome!

**Comment by csnewman** [comment]
Could you rebase these changes ontop of master?

I know I said we were going to use maven, however we'd had to roll that code back. Sorry about the inconvenience.

File: .github/workflows/test1.yml [hunk 1]
```diff
@@ -3,115 +3,138 @@ name: Flutter Engine
...
+      - name: Generate configs
+        run: ${{ format('./linux-gn.sh "{0}"', matrix.config.gn) }}
+      - name: Build
+        run: ${{ format('./macos-build.sh {0}', matrix.config.tag) }}
+      - name: Print tree
+        run: tree ./
+      - name: Uploading artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{ format('engine-linux_x64-{0}', matrix.config.tag) }}
```
Comment (csnewman) [inline .github/workflows/test1.yml]: Can you update the artifact name to reflect that its macos?

Comment (ea2973929) [reply .github/workflows/test1.yml]: Ah, right! Sure thing.

**Comment by ea2973929** [comment]
> Could you rebase these changes ontop of master?
> 
> I know I said we were going to use maven, however we'd had to roll that code back. Sorry about the inconvenience.

Sure, it looks kinda messed up right now.

**Comment by ea2973929** [comment]
> If you have time and interest in getting the ios build working, that would be awesome!

I'll see what I can do. I'll probably go on and finish cargo-flutter for mac os x first. Then tbh my personal next goal would be to get Windows-support, but maybe I can squeeze in iOs somewhere there inbetween.

**Comment by dvc94ch** [comment]
Yeah ios won't work out of the box at the moment, but getting the engine to build would be awesome! Windows support would also be great

**Comment by csnewman** [comment]
LGTM, lets merge this, and then you can follow up with iOS & Windows (each in its own PR).

**Pull request merged**